src/assignment/hooks/assignment-blocks.tsx

```diff
@@ -118,11 +118,60 @@ export function useAssignmentItemBlock(
 							: undefined
 					}
 				>
-					{renderMeta(block)}
+					<AssignmentBlockMeta block={block} />
 				</CustomiseItemForm>
 			);
 		},
-		[t, setBlock, getButtons, renderMeta]
+		[t, setBlock, getButtons]
+	);
+}
+
+export function useAssignmentSearchBlock(
+	setBlock: (block: AssignmentBlock, update: Partial<AssignmentBlock>) => void
+) {
```

Comment:
This is better suited as a component:
```
<AssignmentSearchBlock block={block} setBlock={setBlock} />
```

instead of:
```
const search = useAssignmentSearchBlock(setBlock);

....

return search(block);
```

That way it behaves just like a TextInput:
```
<TextInput value={value} onChange={setValue} />
```